include/cfdcore/cfdcore_ecdsa_adaptor.h

```diff
@@ -0,0 +1,175 @@
+// Copyright 2020 CryptoGarage
+
+#include <string>
+
+#include "cfdcore/cfdcore_bytedata.h"
+#include "cfdcore/cfdcore_common.h"
+#include "cfdcore/cfdcore_key.h"
+
+#ifndef CFD_CORE_INCLUDE_CFDCORE_CFDCORE_ECDSA_ADAPTOR_H_
+#define CFD_CORE_INCLUDE_CFDCORE_CFDCORE_ECDSA_ADAPTOR_H_
+
+namespace cfd {
+namespace core {
+
+using cfd::core::ByteData;
+using cfd::core::ByteData256;
+using cfd::core::Privkey;
+using cfd::core::Pubkey;
+
+/**
+ * @brief An adaptor signature.
+ *
+ */
+class CFD_CORE_EXPORT AdaptorSignature {
+ public:
+  /**
+   * @brief Size of an adaptor signature.
+   *
+   */
+  static constexpr uint32_t kAdaptorSignatureSize = 65;
+  /**
+   * @brief Construct a new Adaptor Signature object from ByteData
+   *
+   * @param data the data representing the adaptor signature
+   */
+  explicit AdaptorSignature(const ByteData &data);
+  /**
+   * @brief Construct a new Adaptor Signature object from a string
+   *
+   * @param data the data representing the adaptor signature
+   */
+  explicit AdaptorSignature(const std::string &data);
+
+  /**
+   * @brief Get the underlying ByteData object
+   *
+   * @return ByteData
+   */
+  ByteData GetData() const;
+
+ private:
+  /**
+   * @brief The underlying data
+   *
+   */
+  ByteData data_;
+};
+
+/**
+ * @brief A proof for an adaptor signature.
+ *
+ */
+class CFD_CORE_EXPORT AdaptorProof {
+ public:
+  /**
+   * @brief The size of an adaptor proof.
+   *
+   */
+  static constexpr uint32_t kAdaptorProofSize = 97;
+  /**
+   * @brief Construct a new Adaptor Proof from ByteData
+   *
+   * @param data the data representing the proof
+   */
+  explicit AdaptorProof(const ByteData &data);
+  /**
+   * @brief Construct a new Adaptor Proof object from a string
+   *
+   * @param data the data representing the proof
+   */
+  explicit AdaptorProof(const std::string &data);
+
+  /**
+   * @brief Get the underlying ByteData object.
+   *
+   * @return ByteData
+   */
+  ByteData GetData() const;
+
+ private:
+  /**
+   * @brief The underlying data
+   *
+   */
+  ByteData data_;
+};
+
+/**
+ * @brief A pair of an adaptor signature together with the proof attached to it.
+ *
+ */
+struct CFD_CORE_EXPORT AdaptorPair {
```

Comment:
The struct does not need to be exported as it will be materialized at the caller.
```suggestion
struct AdaptorPair {
```

Reply:
Changed